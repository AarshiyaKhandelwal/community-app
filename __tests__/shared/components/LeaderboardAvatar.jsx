import React from 'react';
import Renderer from 'react-test-renderer/shallow';
import TU from 'react-dom/test-utils';
import LeaderboardAvatar from 'components/LeaderboardAvatar';

const mockData = {
  member: {
    handle: 'handle',
  },
  domain: 'domain',
  url: 'url',
};

const mockData2 = {
  member: {
    handle: 'handle',
    isSmr: true,
    photoURL: 'photoURL',
  },
  domain: 'domain',
};

const mockData3 = {
  member: {
    handle: 'handle',
    isSmr: true,
    photoURL: 'photoURL',
    position: '1',
  },
  domain: 'domain',
};

test('Matches shallow shapshot', () => {
  const renderer = new Renderer();

  renderer.render((
    <LeaderboardAvatar {...mockData} />
  ));
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('Render properly', () => {
  let instance = TU.renderIntoDocument((<LeaderboardAvatar {...mockData2} />));
  expect(instance.state.member).toEqual(mockData2.member);
  instance.handleError();
  expect(instance.state.member.photoURL).toEqual('https://acrobatusers.com/assets/images/template/author_generic.jpg');

  instance = TU.renderIntoDocument((<LeaderboardAvatar {...mockData3} />));
  expect(instance.state.member).toEqual(mockData3.member);
});
