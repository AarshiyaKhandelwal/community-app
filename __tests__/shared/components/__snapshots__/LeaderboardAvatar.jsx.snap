// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches shallow shapshot 1`] = `
<a
  className="src-shared-components-LeaderboardAvatar-___style__leaderboard-avatar___3GHTf src-shared-components-LeaderboardAvatar-___style__light-gray___2hXZL"
  href="url"
>
  han
  <span
    className="src-shared-components-LeaderboardAvatar-___style__hidden___-vnvD"
  />
</a>
`;
